shared/rollouts/__init__.py

```diff
@@ -80,23 +70,41 @@ class Feature:
 
     HASHSPACE = 2**128
 
-    def __init__(
-        self,
-        name,
-        proportion,
-        variants={"enabled": FeatureVariant(True, 1.0)},
-        salt="",
-        overrides={},
-    ):
-        assert sum(map(lambda x: x.proportion, variants.values())) == 1.0
-        assert proportion >= 0 and proportion <= 1.0
-        self.name = name
-        self.proportion = proportion
-        self.variants = variants
-        self.salt = salt
-        self.overrides = overrides
-
-    @cached_property
+    def __init__(self, name, proportion=None, salt=None):
+        assert not proportion or proportion >= 0 and proportion <= 1.0
+        assert not salt or isinstance(salt, str)
+
+        ff_query = FeatureFlag.objects.filter(pk=name)
+
+        if len(ff_query) == 0:
+            # create default feature flag
+            args = {name: name}
+            if proportion:
+                args["proportion"] = proportion
+            if salt:
+                args["salt"] = salt
+
+            self.feature_flag = FeatureFlag(**args)
+            self.feature_flag.save()
+        else:
+            # ignore constructor args if db entry already exists.
+            self.feature_flag = ff_query.first()
+
+        self.ff_variants = list(
+            FeatureFlagVariant.objects.filter(feature_flag=self.feature_flag.name)
+        )
```

Comment:
you could probably just make this the definition of `_fetch_and_set_from_db()` and reuse that function here